### PR TITLE
Fix transaction history in ABCI++ mode

### PR DESCRIPTION
**Pull request by Scar26**
Fixes: #301 and partially #300

based on #357

### CODE REVIEW

**Comment by Scar26** [comment]
Now covers #300 fully

File: tests/src/e2e/ledger_tests.rs
Comment (tzemanovic) [inline tests/src/e2e/ledger_tests.rs:600]: there's a helper fn imported for this

```suggestion
    let albert_address = find_address(&test, ALBERT)?;
```

File: tests/src/e2e/ledger_tests.rs
Comment (tzemanovic) [inline tests/src/e2e/ledger_tests.rs:605]:
```suggestion
    let bertha_address = find_address(&test, BERTHA)?;
```

**Review by tzemanovic — DISMISSED** [review]
the tests LGTM, thanks! If you could pls also apply `make fmt`

**Comment by tzemanovic** [comment]
we're currently blocked on using this due to an issue with switching on kv indexer in tendermint that is needed to support this transaction history impl. Tendermint indexer crashes with `panic: interface conversion: types.EventData is types.EventDataNewBlock, not types.EventDataNewBlockHeader` when we submit more than 2 txs in parallel, which is caught by the `e2e::ledger_tests::ledger_many_txs_in_a_block` test

**Pull request closed without merging**
